licensekey.dat, ts3server.sqlitedb, updated link, verify hash

ghtyrant opened the pull request:
Hey!

A few changes:

- This formula now manages a licensekey.dat, if provided in pillars.
- The sls_block macro caused a syntax error for me, so I removed it and simplified it to source
- While changing that, I added archive_hash to verify the hash of the downloaded archive
- I also updated the link to an up-to-date version of ts3server
- Install a systemd .service on systems running systemd

ghtyrant committed:
Remove archive macro, update link, add hash

ghtyrant committed:
Manage licensekey.dat via pillars, import sqlitedb on demand

ghtyrant committed:
Use a systemd service on systems running systemd

ghtyrant committed:
Fix id conflict

ghtyrant committed:
Fix permissions on sqlitedb and licensekey.dat

timniederhausen commented:
Looks good, thanks for the contribution!

I'd like to keep the `sls_block` macro though, as it allows passing additional options to the `file.managed` state (e.g. skip_verify). It probably didn't work for you because the default in defaults.yaml was broken (sorry about that).
A correct default would look like this:
```sls
teamspeak:
  archive:
    source: http://dl.4players.de/ts/releases/3.0.13.8/teamspeak3-server_linux_amd64-3.0.13.8.tar.bz2
    source_hash: 460c771bf58c9a49b4be2c677652f21896b98a021d7fff286e59679b3f987a59
```

ghtyrant committed:
Revert "Remove archive macro, update link, add hash"

This reverts commit 718fb12c6a8bdb44b67920a1b5d421fa6e1a2ff1.

ghtyrant committed:
Add updated URL and hash to pillar.example and defaults.yaml

ghtyrant commented:
Yes, the broken defaults.yaml made me rip that out - I'm still very new to SaltStack :)

I've reverted the changes related to the macro and updated defaults and pillar.example.

timniederhausen commented:
Thanks!
